moban/engine_handlebars.py

```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+
+import moban.utils as utils
+import moban.reporter as reporter
+import moban.constants as constants
+import moban.exceptions as exceptions
+from lml.loader import scan_plugins_regex
+from lml.plugin import PluginInfo
+from moban.base_engine import BaseEngine
+from moban.engine_factory import (
+    MOBAN_ALL,
+    BUILTIN_EXENSIONS,
+    Context,
+    expand_template_directory,
+    expand_template_directories,
+    verify_the_existence_of_directories
+)
+
+from pybars import Compiler
+
+
+@PluginInfo(
+    constants.TEMPLATE_ENGINE_EXTENSION, tags=["handlebars", "hbs"]
+)
+class EngineHandlebars(BaseEngine):
+    def __init__(self, template_dirs, context_dirs):
+        BaseEngine.__init__(self)
+        scan_plugins_regex(MOBAN_ALL, "moban", None, BUILTIN_EXENSIONS)
+        template_dirs = list(expand_template_directories(template_dirs))
+        verify_the_existence_of_directories(template_dirs)
+        context_dirs = expand_template_directory(context_dirs)
+        self.context = Context(context_dirs)
+        self.template_dirs = template_dirs
+
+    def find_template_file(self, template_file):
+        for directory in self.template_dirs:
+            if os.path.exists(os.path.join(directory, template_file)):
+                return os.path.abspath(os.path.join(directory, template_file))
+        raise exceptions.FileNotFound(template_file)
+
+    def render_to_file(self, template_file, data_file, output_file):
```

Comment:
the same comment here. sometimes, you are right to question the existing code: why is it done like this? Duplicated code is an obvious violation of DIY rule.